envisage/plugin.py

```diff
@@ -327,7 +327,7 @@ def _anytrait_changed(self, trait_name, old, new):
             else:
                 added = new
                 removed = old
-                index = slice(0, max(len(old), len(new)))
+                index = slice(0, len(old))
```

Comment:
This is a bug fix required for this PR. I should open a separate PR for this.

Reply:
I'm not sure what the bug is here. Can you elaborate?

Reply:
oops, I failed to open that separate PR. I should do that and it would be easier to see it there :)

Reply:
Done: https://github.com/enthought/envisage/pull/357